mission_control/src/mission_control.cpp

```diff
@@ -271,91 +194,94 @@ void MissionControlNode::loadMissionCallback(const mission_control::LoadMission:
   {
     outmsg.load_state = ReportLoadMissionState::FAILED;
     outmsg.mission_id = 0;
-    ROS_INFO("loadMissionCallback - FAILED to parse mission file '%s'",
-             (msg->mission_file_full_path).c_str());
+    ROS_INFO_STREAM("loadMissionCallback - FAILED to parse mission file "
+                    << msg->mission_file_full_path.c_str());
   }
   else
   {
     outmsg.load_state = ReportLoadMissionState::SUCCESS;
     outmsg.mission_id = m_mission_id_counter;
-    ROS_INFO("loadMissionCallback - SUCCESSFULLY parsed mission file '%s'",
-             (msg->mission_file_full_path).c_str());
+    ROS_INFO_STREAM("loadMissionCallback - SUCCESSFULLY parsed mission file "
+                    << msg->mission_file_full_path.c_str());
   }
 
   outmsg.header.stamp = ros::Time::now();
-
   pub_report_mission_load_state.publish(outmsg);
 }
 
 void MissionControlNode::executeMissionCallback(
     const mission_control::ExecuteMission::ConstPtr& msg)
 {
-  // TODO(qna): Need to shutdown any mission that is currently running.
-  ROS_INFO("executeMissionCallback - Executing mission id[%d]", msg->mission_id);
-  executeMission(msg->mission_id);
+  if (m_current_mission_id > 0)
+  {
+    BT::NodeStatus missionStatus = m_mission_map[m_current_mission_id]->getMissionStatus();
+    if ((missionStatus != BT::NodeStatus::IDLE) && (missionStatus != BT::NodeStatus::SUCCESS))
```

Comment:
@patapufate what if the mission failed?

Reply:
The executionTimer will not start the mission and the method will print a message of status=failure.

Reply:
AFAICS the only method that sets `currentMissionId` is this one and the callback for mission removal. If the current mission failed, then the condition of this `if` clause will evaluate to `true`. Thus, you won't be able to execute any other mission until you've removed them all.